specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/virtualWan.json

```diff
@@ -2593,7 +2901,330 @@
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VpnServerConfigRadiusServerRootCertificate"
+          },
+          "description": "Radius Server root certificate of VpnServerConfiguration."
+        },
+        "vpnServerConfigRadiusClientRootCertificates": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VpnServerConfigRadiusClientRootCertificate"
+          },
+          "description": "Radius client root certificate of VpnServerConfiguration."
+        },
+        "vpnClientIpsecPolicies": {
+          "type": "array",
+          "items": {
+            "$ref": "./virtualNetworkGateway.json#/definitions/IpsecPolicy"
+          },
+          "description": "VpnClientIpsecPolicies for VpnServerConfiguration."
+        },
+        "radiusServerAddress": {
+          "type": "string",
+          "description": "The radius server address property of the VpnServerConfiguration resource for point to site client connection."
+        },
+        "radiusServerSecret": {
+          "type": "string",
+          "description": "The radius secret property of the VpnServerConfiguration resource for point to site client connection."
+        },
+        "aadAuthenticationParameters": {
+          "$ref": "#/definitions/AadAuthenticationParameters",
+          "description": "The set of aad vpn authentication parameters."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state of the VpnServerConfiguration resource. Possible values are: 'Updating', 'Deleting', and 'Failed'."
+        },
+        "p2SVpnGateways": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "./network.json#/definitions/SubResource"
+          },
+          "description": "List of references to P2SVpnGateways."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "description": "Parameters for VpnServerConfiguration."
+    },
+    "VpnServerConfiguration": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VpnServerConfigurationProperties",
+          "description": "Properties of the P2SVpnServer configuration."
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "etag": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
```

Comment:
I assume this should be a reference to: "./network.json#/definitions/Resource" and not "SubResource"?  Taking "SubResource" into account, this is missing: `type`, `tags`, and `location`.

Reply:
Certificates resources e.g. VpnServerConfigVpnClientRootCertificate are child resources of top level resource: VpnServerConfiguration. Isn't this correct then?